### ISSUE TITLE
If a query is applied without containing the_geom or the_geom_webmercator the maps API fails

### ISSUE THREAD

**Issue by xavijam** [CartoDB/cartodb#7876]
### STR

- Create a map with a points dataset.
- Go to the builder.
- Go to the SQL editor again, and apply this query: `SELECT cartodb_id FROM xxxxx`.
- Query is ok but Map fails because there is no geometry. 
  - `Request to Maps API failed: Postgis Plugin: ERROR:  column "the_geom_webmercator" does not exist LINE 1: SELECT ST_AsBinary(ST_Simplify(ST_SnapToGrid("the_geom_webme...`

**Comment by xavijam** [comment]
The question here, do we want this behaviour? is it correct? If so, I'll do the proper changes in my implementation.

**Comment by viddo** [comment]
Actually that works correctly/as it was intended; if the query is changed the query-schema model _should_ be set to `unfetched` because if the query changed the schema might have changed too, so the schema needs to be reset and re-fetched. 
Even more so if editing the query of a source node that have an analysis on top of it, because the query change might affect each node's schema in that chain.

The alternative would be interpret the changed query to decide whether it changes the schema or not, but that could easily become complicated… not sure we want to go down that path.

**Comment by xavijam** [comment]
No no, that's not the problem. If you check the steps, if there is any problem at map level (cc @alonsogarciapablo) after applying a query (query is ok), the query schema model status is reseted to 'unfetched'.

**Comment by viddo** [comment]
Like mentioned offline; Since a source node doesn't really need the result from maps API to update the query-schema model this could/should (probably) be handled by the source-node model instead. So in terms of changes we'd have to: 
- Update/reset/re-fetch query-schema-model in [source-node-model](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb3/data/analysis-definition-node-source-model.js) when the query is changed.
- Don't listen to [node changes in deep-insights-integration](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb3/deep-insights-integrations.js#L69-L77) if the node is of type source

**Comment by viddo** [comment]
This report actually contains 2 bugs, the first one related to query-schema model was resolved by #8232. 

However, setting a query that doesn't include the_geom makes the maps API fail yeah:

```
but the maps API still fails:
```
Request to Maps API failed: Postgis Plugin: ERROR:  column "the_geom_webmercator" does not exist
LINE 1: SELECT ST_AsBinary(ST_Simplify(ST_SnapToGrid("the_geom_webme...`)
```

Updating the title+desc to reflect this, but I'm not sure how we should handle/resolve that really.

**Comment by viddo** [comment]
Closing since there's nothing more to do here, side-effects of this kind of query will be handled like a non-select query which is planned for #8093
